flang/include/flang/Lower/AbstractConverter.h

```diff
@@ -123,6 +123,11 @@ class AbstractConverter {
   /// Generate the location as converted from a CharBlock
   virtual mlir::Location genLocation(const Fortran::parser::CharBlock &) = 0;
 
+  /// Generate a string literal containing the file name and return its address
+  virtual mlir::Value locationToFilename(mlir::Location) = 0;
+  /// Generate a constant of the given type with the location line number
+  virtual mlir::Value locationToLineNo(mlir::Location, mlir::Type) = 0;
```

Comment:
These functions already exist in IO.cpp. The ones there can be moved to the converter.

Reply:
Right, it's actually coming from there, I did not push the IO change. Fixed.